src/components/tree/tree.ts

```diff
@@ -0,0 +1,248 @@
+import { LitElement, html } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import { clamp } from 'src/internal/math';
+import { watch } from 'src/internal/watch';
+import { isTreeItem } from '../tree-item/tree-item';
+import styles from './tree.styles';
+import type SlTreeItem from '../tree-item/tree-item';
+
+function syncAncestors(treeItem: SlTreeItem) {
+  const parentItem: SlTreeItem | null = treeItem.parentElement as SlTreeItem;
+
+  if (isTreeItem(parentItem)) {
+    const children = parentItem.getChildrenItems({ includeDisabled: false });
+    const allChecked = children.every(item => item.selected);
+    const allUnchecked = children.every(item => !item.selected && !item.indeterminate);
+
+    parentItem.setSelected(allChecked);
+    parentItem.indeterminate = !allChecked && !allUnchecked;
+
+    syncAncestors(parentItem);
+  }
+}
+
+function syncDescendants(treeItem: SlTreeItem) {
+  for (const childItem of treeItem.getChildrenItems()) {
+    childItem.setSelected(!childItem.disabled && treeItem.selected);
+    syncDescendants(childItem);
+  }
+}
+
+/**
+ * @since 2.0
+ * @status experimental
+ *
+ * @dependency sl-tree-item
```

Comment:
I believe we can remove this. We only need to label dependencies that are used within a component's template. Since `<sl-tree-item>` is slotted in by the user, they're expected to import/register it themselves.

Reply:
fixed [6413915](https://github.com/shoelace-style/shoelace/pull/823/commits/641391589df9c9fbd8c598caf6e8e5afe21a1a71)